hadoop-hdds/interface-client/src/main/proto/hdds.proto

```diff
@@ -479,4 +479,19 @@ message DeletedBlocksTransactionInfo {
     optional int64 containerID = 2;
     repeated int64 localID = 3;
     optional int32 count = 4;
-}
\ No newline at end of file
+}
+
```

Comment:
Can we have sub proto with both start key and end key put together? Either this node has both values or none?
```suggestion
    message FileInfoRange {
    required string startKey = 2;
    required string endKey = 3;
}
message CompactionFileInfoProto {
    required string fileName = 1;
    optional FileInfoRange range = 2;
    optional string columnFamily = 3;
}
```

Reply:
Technically `startKey`, `endKey` and `columnFamily` are required but there are optional to make it backward compatible.

What is the benefit of creating separate proto message `FileInfoRange`? What's your rational behind it?

Reply:
As in it makes sense to have either both the values set or none. It shouldn't be none. For some reason if there is a bug, we might end up with a wrong meta.

Reply:
If any of `startKey`, `endKey` and `columnFamily` is null, we should not be using it to filter the DAG because if `columnFamily` is null, `startKey` and `endKey` doesn't matter.

Something similar to https://github.com/apache/ozone/pull/5236/files#diff-1d5c6c892a2d24d85ca002dafcb9f39341192baaa7b5323d2325b5fc7a238c7fR1598

So basically it should be on the caller of `CompactionFileInfo` and decide how to handle null case scenario.

Reply:
If you really want it to be safe, I can add a condition in the constructor of [CompcationFileInfo](https://github.com/apache/ozone/pull/5303/files#diff-f6807cfb3d601de6e4d002178c53f17767c9547ae04d19c16e089ececaea10a3R26) to make sure that either all three are present or none but creating a another class for it is unnecessary nesting to me.

Reply:
yeah that makes sense adding a non null check on a constructor makes sense.

Reply:
Updated the construction of `CompactionFileInfo`.